libs/sdk-backend-tiger/src/convertors/toBackend/afm/TotalsConverter.ts

```diff
@@ -0,0 +1,114 @@
+ * tiger AFM. Totals in tiger are defined more explicitly and therefore also more generally. Specifically a total is
+ * defined by `includedDimensions`, i.e. the actual dimension on which it will be computed. Also there's a possibility
+ * to compute totals only for some attribute values from an attribute in a dimension. While this is again working
+ * for any attribute (e.g. compute totals only for US column of Country attribute) it's probably mostly useful only in
+ * case of the measureGroup pseudo-attribute.
+ *
+ * TODO: subtotals
+ */
+export function convertTotals(def: IExecutionDefinition): Array<GrandTotal> {
+    const allDimensionIndexes = def.dimensions.map((_dim, idx) => idx);
+    return withTotals(def.dimensions, (dimIdx, _typeIdx, totalsOfType) => {
+        // (one-dimensional) grand total is defined in Tiger by all dimensions except the current one
+        const includedDimensionIndexes = allDimensionIndexes.filter((idx) => dimIdx != idx);
+        const measureIdentifiers = totalsOfType.map((total) => total.measureIdentifier);
+        const includedDimensions = includedDimensionIndexes.map((includedDimIdx) => {
+            const dim = def.dimensions[includedDimIdx];
+            const dimensionAttributesValues = dim.itemIdentifiers.includes(MeasureGroupIdentifier)
+                ? [{ itemIdentifier: MeasureGroupIdentifier, values: measureIdentifiers }]
+                : null;
+            // FIXME synchronize dimensionIdentifier naming with convertDimensions
+            return { dimensionIdentifier: `dim_${includedDimIdx}`, dimensionAttributesValues };
+        });
+        const totalType = totalsOfType[0].type;
+
+        return {
+            localIdentifier: totalLocalIdentifier(totalType, dimIdx),
+            function: convertTotalType(totalType as TotalType),
+            includedDimensions,
+        };
+    });
+}
+
+/**
+ * Traverse given dimensions and their total definitions, group those by total type and call totalProcessor on each
+ * of these groups, together with current dimension index (dimIdx) and total type index in that dimension (typeIdx).
+ *
+ * This function captures the contract between execution response and execution result. Both of these response
+ * structures are synchronized using local identifiers induced from dimIdx typeIdx. In the future the function should
+ * be replaced by explicit concept of total local identifiers both in ITotal and ITotalDescriptor. Currently, given
+ * the rather limit functionality of general totals API, it's needed/useful only in Tiger.
+ */
+export function withTotals<T>(
+    dimensions: IDimension[],
+    totalProcessor: (dimIdx: number, typeIdx: number, totalsOfType: ITotal[]) => T,
+): T[] {
+    validateTotals(dimensions);
+    return flatMap(dimensions, (dim, dimIdx) => {
+        const totalsByType = groupBy(dim.totals || [], (total) => total.type);
+        return Object.keys(totalsByType).map((type, typeIdx) =>
```

Comment:
please change Object.keys() to `keys` function from lodash. signature is identical. the thing is Object.keys() may not be available in all browsers.